File: rb/features.rb
# RemoteJobsApiRssFeed SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module RemoteJobsApiRssFeedFeatures
  def self.make_feature(name)
    case name
    when "base"
      RemoteJobsApiRssFeedBaseFeature.new
    when "ratelimit"
      RemoteJobsApiRssFeedRatelimitFeature.new
    when "retry"
      RemoteJobsApiRssFeedRetryFeature.new
    when "test"
      RemoteJobsApiRssFeedTestFeature.new
    when "timeout"
      RemoteJobsApiRssFeedTimeoutFeature.new
    else
      RemoteJobsApiRssFeedBaseFeature.new
    end
  end
end

File: py/features.py
# RemoteJobsApiRssFeed SDK feature factory

from feature.base_feature import RemoteJobsApiRssFeedBaseFeature
from feature.test_feature import RemoteJobsApiRssFeedTestFeature


def _make_feature(name):
    features = {
        "base": lambda: RemoteJobsApiRssFeedBaseFeature(),
        "test": lambda: RemoteJobsApiRssFeedTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
